[docs] Update CSS theme variables related content (#43130)
mui/material-ui: docs/data/material/migration/migrating-to-v6/migrating-to-v6.md

## Stabilized APIs

### Merged CssVarsProvider into ThemeProvider

> The `CssVarsProvider` and `extendTheme` has been merged into `ThemeProvider` and `createTheme`.

Reviewer: Maybe:

```suggestion
The `CssVarsProvider` and `extendTheme` features have been ported into `ThemeProvider` and `createTheme`.
```

If you are using them in v5 or v6-beta, you must migrate as shown below:

```diff

> -import { experimental_extendTheme as extendTheme, Experimental_CssVarsProvider as CssVarsProvider } from '@mui/material/styles';

Reviewer: Let's update the diff to use `createTheme` and `CssVarsProvider` instead: 

```diff
- import { experimental_extendTheme as extendTheme, Experimental_CssVarsProvider as CssVarsProvider } from '@mui/material/styles';
+ import { createTheme, ThemeProvider } from '@mui/material/styles';

- const theme = extendTheme(
+ const theme = createTheme(
     {
+      cssVariables: true,
       // the rest of your theme
     }
);

 function App() {
   return
-    <CssVarsProvider>
+    <ThemeProvider>
       {/* ...you app */}
-    <CssVarsProvider>
+    </ThemeProvider>;
 }
```

Reviewer: I implemented the solution cc: @siriwatknp
